lib/ansible/modules/remote_management/hpilo/hpilo_facts.py

```diff
@@ -167,8 +187,25 @@ def main():
     login = module.params['login']
     password = module.params['password']
     ssl_version = getattr(hpilo.ssl, 'PROTOCOL_' + module.params.get('ssl_version').upper().replace('V', 'v'))
-
-    ilo = hpilo.Ilo(host, login=login, password=password, ssl_version=ssl_version)
+    validate_certs = module.params['validate_certs']
+    ssl_ignore_hostname = module.params['ssl_ignore_hostname']
+    ca_certificate = module.params['ca_certificate']
+
+    if float(hpilo.__version__) >= 4:
```

Comment:
python-hpilo.__version__ is a string, resulting in an error. The cast doesn't work, since the second decimal point renders it as an invalid floating point number:

```
if float(hpilo.__version__) >= 4:
ValueError: invalid literal for float(): 4.2.1
```

FWIW...
```
Python 2.7.14 (default, Dec 28 2017, 13:06:26) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import hpilo
>>> hpilo.__version__
'4.2.1'
>>> type(hpilo.__version__)
<type 'str'>
>>> float(hpilo.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for float(): 4.2.1
```

Reply:
Perhaps a little messier...

````
if float(".".join(hpilo.__version__.split(".")[0:2])) >= 4:
````